src/main/java/Duke.java

```diff
@@ -1,10 +1,45 @@
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("Duke.txt").run();
+    }
+
+    /**
+     * Main body of Duke program.
+     * Command is created by parsing the input and then executed.
+     * Error message is shown if there is DukeException.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```

Comment:
While isExit is a valid variable name, I think you can choose a better one like shouldExit or canExit. From an English readability point of view, it will help things slightly.

Reply:
Thanks for the input.